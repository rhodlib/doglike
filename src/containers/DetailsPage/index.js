import React, { useEffect } from "react";
import { useParams } from "react-router-dom";
import DogDetails from "../../components/DogDetails";
import styles from "./DetailsPage.module.css";
import { useDispatch, useSelector } from "react-redux";
import { getDogImage } from "../../core/actions/racesActions";

/**
 * Component that receives a string and makes a request.
 */
export const DetailsPage = () => {
  const dispatch = useDispatch();
  const { dogname = "" } = useParams();
  const dogRes = useSelector(state => state.raceReducer);

  // Asynchronous request that returns the path of an image.
  useEffect(() => {
    dispatch(getDogImage(dogname));
  }, [dispatch, dogname]);

  return (
    <div className={styles.DetailsPageContainer}>
      <DogDetails
        dogImg={dogRes.img}
        dogName={dogname}
        onReload={() => dispatch(getDogImage(dogname))}
      />
    </div>
  );
};

export default DetailsPage;
