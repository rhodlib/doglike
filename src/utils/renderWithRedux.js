import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import reducers from "../core/reducers";
import { initialState as defaultInitialState } from "../core";
import { render } from "@testing-library/react";
import reduxThunk from "redux-thunk";

/**
 * This is a Util to render test with redux.
 */
export const renderWithRedux = (
  ui,
  {
    initialState = defaultInitialState,
    store = createStore(reducers, initialState, applyMiddleware(reduxThunk))
  } = {}
) => ({ ...render(<Provider store={store}>{ui}</Provider>), store });

export default renderWithRedux;
