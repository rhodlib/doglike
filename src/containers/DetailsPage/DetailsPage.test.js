import React from "react";
import DetailsPage from "./index";
import renderWithRedux from "../../utils/renderWithRedux";
import { BrowserRouter as Router } from "react-router-dom";

it("Matches snapshot", () => {
  const { asFragment } = renderWithRedux(
    <Router>
      <DetailsPage />
    </Router>,
    { initialState: { raceReducer: { img: "test" } } }
  );
  expect(asFragment()).toMatchSnapshot();
});
