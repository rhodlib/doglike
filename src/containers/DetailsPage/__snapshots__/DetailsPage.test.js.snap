// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="DetailsPageContainer"
  >
    <div
      class="DogDetails"
    >
      <div
        class="Loader Spinner"
      />
    </div>
  </div>
</DocumentFragment>
`;
